clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -219,30 +219,24 @@ def drain_node(self, node_name: str, disable_eviction: bool) -> bool:
             return False
 
     def cordon_node(self, node_name: str) -> bool:
+        now = str(arrow.now().timestamp)
+        key = CLUSTERMAN_TERMINATION_TAINT_KEY
         try:
-            self._core_api.patch_node(
-                name=node_name,
-                body={
-                    "spec": {
-                        "unschedulable": True,
-                    },
-                },
-            )
+            node = self._core_api.read_node(node_name)
```

Comment:
Won't this information be available in `self._nodes_by_ip`? So you don't need to do an extra read.

Reply:
I was thinking to get information real-time to minimise race condition, but yeah agree with you. I'm gonna change it and read real-time in case of necessity.